SimplifiedTests/Mocks/NYPLAnnotationsMock.swift

```diff
@@ -123,24 +125,23 @@ class NYPLAnnotationsMock: NYPLAnnotationSyncing {
     completion(bookmarksUpdated, bookmarksFailedToUpdate)
   }
   
-  static func postBookmark(_ bookmark: NYPLReadiumBookmark,
-                           forBookID bookID: String,
-                           completion: @escaping (_ serverID: String?) -> ()) {
+  static func postBookmark(_ bookmark: NYPLBookmark, forBookID bookID: String, completion: @escaping (String?) -> ()) {
     guard !failRequest else {
       completion(nil)
       return
     }
-    
+
     if let bookmarks = serverBookmarks[bookID],
-       bookmarks.contains(bookmark)
+       bookmarks.contains(where: { $0.annotationId == bookmark.annotationId })
```

Comment:
This change happens because the `NYPLAnnotations` function is now generic using `NYPLBookmark`, and `NYPLBookmark` is a protocol which cannot conform to `Equatable`. We can cast the `bookmark` as `NYPLReadiumBookmark` or `NYPLAudiobookBookmark` to do the comparison if we needed, but for this function I believe checking `annotationId` is good enough because that's all we needed to know if a bookmark is uploaded or not. 
I think we can also resolve this by changing `NYPLBookmark` to a class and make it conform to the `Equatable` protocol. I didn't do it because this is a mock class and in reality the check is performed in the business logic (which `NYPLReadiumBookmark` and `NYPLAudiobookBookmark` are equatable) but not in `NYPLAnnotations`.

Reply:
ohh it's a mock class. Sorry I didn't see it. In any case what you said makes sense.